src/main/java/manufacturing/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,55 @@
+package manufacturing.dao.impl;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import manufacturing.dao.ManufacturerDao;
+import manufacturing.db.Storage;
+import manufacturing.lib.Dao;
+import manufacturing.model.Manufacturer;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> getById(Long manufacturerId) {
+        return Optional.ofNullable(Storage.manufacturers.get(manufacturerId.intValue()));
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Manufacturer manufacturerForUpdate = Storage.manufacturers
+                .stream()
+                .filter(m -> Objects.equals(m.getId(), manufacturer.getId()))
+                .findFirst()
+                .get();
```

Comment:
if I remember correctly, you don't need .get() as findFirst() does the same

Reply:
than it returns Optional instead of Manufacturer object